AmplifyPlugins/DataStore/Tests/DataStoreHostApp/AWSDataStorePluginIntegrationTests/DataStoreObserveQueryTests.swift

```diff
@@ -535,96 +600,95 @@ class DataStoreObserveQueryTests: SyncEngineIntegrationTestBase {
     func testObserveQueryShouldResetOnDataStoreClear() async throws {
         await setUp(withModels: TestModelRegistration())
         try await startAmplifyAndWaitForReady()
-        let firstSnapshotWithIsSynced = expectation(description: "query snapshot with isSynced true")
+        let firstSnapshotWithIsSynced = asyncExpectation(description: "query snapshot with isSynced true")
         var onComplete: ((Subscribers.Completion<Error>) -> Void) = { _ in }
         Amplify.Publisher.create(Amplify.DataStore.observeQuery(for: Post.self))
             .sink { onComplete($0) } receiveValue: { querySnapshot in
                 if querySnapshot.isSynced {
-                    firstSnapshotWithIsSynced.fulfill()
+                    Task { await firstSnapshotWithIsSynced.fulfill() }
                 }
             }.store(in: &cancellables)
-        await waitForExpectations(timeout: 100)
+        await waitForExpectations([firstSnapshotWithIsSynced], timeout: 100)
         
-        let observeQueryReceivedCompleted = expectation(description: "observeQuery received completed")
-        observeQueryReceivedCompleted.isInverted = true
+        let observeQueryReceivedCompleted = asyncExpectation(description: "observeQuery received completed",
+                                                             isInverted: true)
         onComplete = { completed in
             switch completed {
             case .finished:
-                observeQueryReceivedCompleted.fulfill()
+                Task { await observeQueryReceivedCompleted.fulfill() }
             case .failure(let error):
                 XCTFail("\(error)")
             }
         }
         try await Amplify.DataStore.clear()
-        await waitForExpectations(timeout: 10)
+        await waitForExpectations([observeQueryReceivedCompleted], timeout: 10)
     }
 
     func testObserveQueryShouldStartOnDataStoreStart() async throws {
```

Comment:
`testObserveQueryShouldStartOnDataStoreStart` this test can now be updated back to the previous structure, no need to reassign `onSnapshot` now that we have AsyncExpectation, see V1 test for the structure. you can update this test or directly translate the V1 test back to using AsyncExpectation and  callback APIs to async APIs
<img width="695" alt="image" src="https://user-images.githubusercontent.com/1365977/193641814-91ce8f88-a438-493b-8fa7-b6a83c545c5b.png">